pkg/deployment/reconcile/plan_executor.go

```diff
@@ -396,18 +396,30 @@ func (d *Reconciler) executeAction(ctx context.Context, planAction api.Action, a
 		log.Warn("Action aborted. Removing the entire plan")
 		d.context.CreateEvent(k8sutil.NewPlanAbortedEvent(d.context.GetAPIObject(), string(planAction.Type), planAction.MemberID, planAction.Group.AsRole()))
 		return false, true, false, false, nil
-	} else if !timeout.Infinite() {
-		if time.Now().After(planAction.CreationTime.Add(timeout.Duration)) {
-			log.Warn("Action not finished in time. Removing the entire plan")
-			d.context.CreateEvent(k8sutil.NewPlanTimeoutEvent(d.context.GetAPIObject(), string(planAction.Type), planAction.MemberID, planAction.Group.AsRole()))
-			return false, true, false, false, nil
-		}
+	} else if isActionTimeout(timeout, planAction) {
+		log.Warn("Action not finished in time. Removing the entire plan")
+		d.context.CreateEvent(k8sutil.NewPlanTimeoutEvent(d.context.GetAPIObject(), string(planAction.Type), planAction.MemberID, planAction.Group.AsRole()))
+		return false, true, false, false, nil
+
 	}
 
 	// Timeout not yet expired, come back soon
 	return false, false, true, false, nil
 }
 
+func isActionTimeout(timeout api.Timeout, planAction api.Action) bool {
+	if planAction.StartTime == nil {
+		return false
+	}
+	if planAction.StartTime.IsZero() {
+		return false
+	}
+	if timeout.Infinite() {
+		return false
+	}
+	return time.Now().After(planAction.StartTime.Add(timeout.Duration))
```

Comment:
you can use `time.Since(planAction.StartTime) > timeout.Duration`

Reply:
👍